File: MoniGuardAPI/Photo.cs

using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using Microsoft.Extensions.Primitives;

namespace MoniGuardAPI;

public class Photo(int photoId, int cameraId, DateTime createdAt, byte[] content, string name)
{
    public Photo(int cameraId, byte[] content, string name): this(0, cameraId, DateTime.Now, content, name) { }


    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int PhotoId { get; set; } = photoId;

    [ForeignKey("Camera")] 
    public int CameraId { get; set; } = cameraId;

    [Required]
    public DateTime CreatedAt { get; set; } = createdAt;
    
    [JsonIgnore]
    [Required] 
    public byte[]? Content { get; set; } = content;
    // {server ip}/xxx/{sceneId}/{md5}.jpg

    [MaxLength(255)] 
    [Required] 
    public string Name { get; set; } = name;
}

File: moni_guard_api/photo.py
class Photo:
    def __init__(self, photo_id: int, camera_id: int, created_at: str, name: str):
        """
        :param photo_id: 照片 ID。
        :param camera_id: 拍摄摄像机 ID。
        :param created_at: 创建时间。
        :param name: 照片名字。
        """
        self.photo_id = photo_id
        self.camera_id = camera_id
        self.created_at = created_at
        self.name = name


    def __str__(self):
        return f"Photo(photo_id={self.photo_id}, camera_id={self.camera_id}, created_at={self.created_at}, name={self.name})"

    def __repr__(self):
        return str(self)
